parser/src/main/java/me/vertretungsplan/parser/IphisParser.java

```diff
@@ -0,0 +1,358 @@
+        if (matcher.find()) {
+            retArray = matcher.group(1).split(",");
+        }
+        return retArray;
+    }
+
+    private Substitution getSubstitution(JSONObject change, HashMap<String, String> gradesHashMap,
+                                         HashMap<String, String> teachersHashMap)
+            throws IOException, JSONException {
+        final Substitution substitution = new Substitution();
+        // Set class(es)
+        final String[] classIds = getSQLArray(change.getString("id_klasse"));
+        if (classIds.length > 0) {
+            if (gradesHashMap == null) {
+                throw new IOException("Change references a grade but grades are empty.");
+            }
+            final HashSet<String> classes = new HashSet<>();
+            for (String classId : classIds) {
+                classes.add(gradesHashMap.get(classId));
+            }
+            substitution.setClasses(classes);
+        }
+        // Set type
+        final String type = change.getString("aenderungsgrund");
+        substitution.setType(type);
+        // Set color
+        substitution.setColor(colorProvider.getColor(type));
+        // Set covering teacher
+        final String coveringTeacherId = change.getString("id_person_verantwortlich");
+        if (!coveringTeacherId.equals("0")) {
+            if (teachersHashMap == null) {
+                throw new IOException("Change references a covering teacher but teachers are empty.");
+            }
+            substitution.setTeacher(teachersHashMap.get(coveringTeacherId));
+        }
+        // Set teacher
+        final String teacherId = change.getString("id_person_verantwortlich_orig");
+        if (!teacherId.equals("{0}")) {
+            if (teachersHashMap == null) {
+                throw new IOException("Change references a teacher but teachers are empty.");
+            }
+            if (type.equals("Vertretung") || !coveringTeacherId.equals("0")) {
+                substitution.setPreviousTeacher(teachersHashMap.get(teacherId));
+            } else {
+                substitution.setTeacher(teachersHashMap.get(teacherId));
+            }
+
+        }
+        substitution.setDesc(change.getString("information"));
+
+        final String startingHour = change.getString("zeit_von").replaceFirst("^0+(?!$)", "");
+        final String endingHour = change.getString("zeit_bis").replaceFirst("^0+(?!$)", "");
+        if (!startingHour.equals("") || !endingHour.equals("")) {
+            String lesson = "";
+            if (!startingHour.equals("") && endingHour.equals("")) {
+                lesson = "Ab " + startingHour;
+            }
+            if (startingHour.equals("") && !endingHour.equals("")) {
+                lesson = "Bis " + endingHour;
+            }
+            if (!startingHour.equals("") && !endingHour.equals("")) {
+                lesson = startingHour + " - " + endingHour;
+            }
+            substitution.setLesson(lesson);
+        }
```

Comment:
Die `lesson` ist normalerweise die Nummer der Schulstunde ("1" oder "1 - 2" o.ä.) und nicht die Uhrzeit. Wäre das hier auch möglich? Für zwei komplette Uhrzeiten ist das Feld in der App wegen der dort verwendeten großen Schriftgröße zu kurz.

Reply:
hm, gut zu wissen. In aller Regel sollten auch die Schulstunden reichen, es gibt allerdings auch einige Schulen, die haben mehrere Zeitraster... (deshalb haben wir überall die genauen Zeiten mit drin)